src/lib/util/options.h

```diff
@@ -34,41 +34,48 @@ const int OPTION_PRIORITY_MAXIMUM   = 255;          // maximum priority
 
 struct options_entry;
 
-// avoid obnoxious DirectSound issue
-#ifdef ERROR
-#undef ERROR
-#endif
-
 // exception thrown by core_options when an illegal request is made
-class options_exception
+class options_exception : public std::exception
 {
 public:
-	enum class condition_type
-	{
-		NONE,
-		WARNING,
-		ERROR
-	};
+	const std::string &message() const { return m_message; }
+	virtual const char *what() const noexcept override { return message().c_str(); }
+
+protected:
+	options_exception(std::string &&message);
 
-	template <typename... Params>   options_exception(condition_type condition, const char *fmt, Params &&...args)
-		: m_condition(condition)
-		, m_message(util::string_format(fmt, std::forward<Params>(args)...))
+private:
+	std::string m_message;
+};
+
+class options_warning_exception : public options_exception
+{
+public:
+	template <typename... Params> options_warning_exception(const char *fmt, Params &&...args)
```

Comment:
Won't forwarding all arguments cause problems for the overload that takes just `std::string &&message`?

I'm open to the argument that I shouldn't have a ctor that forwards to `util::string_format`